packages/adapter-vercel/index.js

```diff
@@ -544,6 +549,28 @@ async function create_function_bundle(builder, entry, dir, config) {
 		)
 	);
 
+	const server_assets = builder.getServerAssets();
+	let routes_assets = new Set(server_assets.rootErrorPage);
+
+	const routes_length = routes.length;
+	for (let i = 0; i < routes_length; i++) {
```

Comment:
I prefer for…of as well but was worried performance would be an issue. Apparently, this pattern performs much faster so I used it widely in this PR just in case there are lots of routes / assets that need to be iterated over. I can change all of them to for…of if we value the readability more

Reply:
Oh interesting - honestly I would say that we err on performance for hot code paths. I don't know how hot this is, so I guess it's either/or really. I'll leave it up to you!